test-framework/core/src/main/java/org/keycloak/testframework/https/ManagedCertificates.java

```diff
@@ -54,11 +56,38 @@ public ManagedCertificates(CertificatesConfigBuilder configBuilder) throws Manag
         initServerCerts(configBuilder.getKeystoreFormat());
     }
 
+    public SSLContext getClientSSLContext() {
+        try {
+            return SSLContextBuilder.create()
+                    .loadTrustMaterial(clientsTrustStore, null)
+                    .build();
+        } catch (KeyManagementException | NoSuchAlgorithmException | KeyStoreException e) {
+            throw new ManagedCertificatesException(e);
+        }
+    }
+
+    public SSLContext getMTlsClientSSLContext() {
```

Comment:
I'd combine these into a single method; and have it return the SSLContext as needed. We could also move the TLS enabled to within managed certificates; and it it's not enabled then we just return null here; then the admin client can just getSSLContext from ManagedCertificates and if not null then use it.